runtime/CMakeLists.txt

```
@@ -26,9 +27,33 @@ set(backend_includes "${PROJECT_SOURCE_DIR}/lib/backend/common")
 set(QIRSTDLIB_VERSION "0.4.0")
 set(QIRSTDLIB_REPO "https://github.com/qir-alliance/qir-runner")
 
+# Get LLVM hash to target from source tree.
+file(READ ../.dep-versions DEPENDENCY_VERSIONS)
+string(REGEX MATCH "llvm=([0-9a-f]+)" _ ${DEPENDENCY_VERSIONS})
+set(LLVM_HASH ${CMAKE_MATCH_1})
+message(STATUS "Detected LLVM version - ${LLVM_HASH}")
+
+FetchContent_Declare(
+    MLIRRunnerUtils
+    URL                 https://raw.githubusercontent.com/llvm/llvm-project/${LLVM_HASH}/mlir/include/mlir/ExecutionEngine/RunnerUtils.h
+    DOWNLOAD_NO_EXTRACT True
+    SOURCE_DIR        mlir/ExecutionEngine
+)
+
+FetchContent_Declare(
+    MLIRCRunnerUtils
+    URL                 https://raw.githubusercontent.com/llvm/llvm-project/${LLVM_HASH}/mlir/include/mlir/ExecutionEngine/CRunnerUtils.h
+    DOWNLOAD_NO_EXTRACT True
+    SOURCE_DIR          mlir/ExecutionEngine
+)
+
+if(RUNTIME_ENABLE_WARNINGS)
+    message(STATUS "Building with compiler warnings as errors enabled.")
+    add_compile_options(-Werror -Wextra -Wall)
```

Building Kokkos with `-Wextra` is the origin of the build failures in CIs.
```suggestion
    add_compile_options(-Werror -Wall)
```

The other way around is by silencing `-Werror=missing-field-initializers` (there're more errors to be silenced to get all CI jobs pass) with the same trick you used for `catalyst_qir_qis_obj`.

---

Thank you 🙏